synspec.py

```diff
@@ -0,0 +1,50 @@
+from synphot.observation import Observation
+import astropy.units as u
+import numpy as np
+
+
+def from_R(source_spectrum, bandpass, R, waverange, force='none'):
+    """
+    Returns a spectrum with bins defined by a spectral resolution
+    R and a given range of wavelengths. The bin width varies over the
+    range of wavelengths such that the ratio of the wavelength to the
+    bin width at that wavelength is equal to the given R.
+    """
+    first_wl = waverange[0].to(u.angstrom).value
+    binset = np.array([first_wl])
+    binwidth = first_wl / R
+
+    while binset[-1] <= waverange[-1].to(u.angstrom).value:
+        next_wl = binset[-1] + binwidth / 2
+        binset = np.append(binset, next_wl)
+        binwidth = next_wl / R
+
+    return Observation(source_spectrum, bandpass, binset=binset * u.angstrom,
+                       force=force)
+
+
+def from_wave_binwidth(source_spectrum, bandpass, binwidth, waverange=None,
+                       force='none'):
+    """
+    Returns a spectrum with bins defined by a constant bin width specified
+    by the user.
+    """
+    if not waverange:
```

Comment:
I think it's a little clearer what you mean if you write this as `if waverange is None:`